uint32 module_count
float32 wheel_radius